openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -55,18 +58,16 @@ def __init__(self, course_key, request, username=''):
             username or request.user.username,
             course_key,
         )
-        self.effective_user = self.overview.effective_user
         self.original_user_is_staff = has_access(request.user, 'staff', self.overview).has_access
         self.course_key = course_key
         self.enrollment_object = CourseEnrollment.get_enrollment(self.effective_user, self.course_key,
                                                                  select_related=['celebration'])
-        course_masquerade, user = setup_masquerade(
+        self.course_masquerade, self.effective_user = setup_masquerade(
             request,
             course_key,
             staff_access=self.original_user_is_staff,
         )
-        self.is_staff = has_access(user, 'staff', self.overview).has_access
-        self.course_masquerade = course_masquerade
+        self.is_staff = has_access(self.effective_user, 'staff', self.overview).has_access
```

Comment:
:+1: 

`self.is_staff` is still determined based on the user user returned from `setup_masquerade`, we're just pulling it via a different variable name here.